### PR TITLE
feat: Refactor Meroxa-py to handle API Changes

### PR DESCRIPTION
**Pull request by ericcheatham**
# Description

**This is a breaking change that needs to be paired with an update in turbine-py** 

`meroxa-py` is not currently resilient to changes in the API response. This is due to the fact that when `meroxa-py` gets a response back from `platform-api` it attempts to deserialize the response into a `Class` with a specific structure. If that class is missing (or expecting) a particular field to be there when it is not (or is)  the json decode will fail and `meroxa-py` will pass an error to `turbine-py`. 

This change will instead pass the JSON directly out and let turbine-py figure out what it would like to do with it. 

### Change summary
- Return `JSON` response bodies instead of ClassObjects 
   - Class methods return one value now as opposed to a tuple of optionals
   - Removed decorators for response formatting
   - Removed Custom response class objects
  

fixes: https://github.com/meroxa/meroxa-py/issues/52

## Type of change

Please delete options that are not relevant.

- [ ] New feature
- [x] Bug fix
- [x] Refactor
- [ ] Documentation

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] Unit Tests
- [x] Manual Tests
- [ ] Deployed to staging

# Additional references

*Any additional links (if appropriate)*

### CODE REVIEW

**Comment by ericcheatham** [comment]
I still need to test this with turbine-py and make any subsequent changes.

File: tests/test_resources.py [hunk 1]
```diff
@@ -130,14 +73,14 @@ async def test_resources_delete_success(mock_session):
...
-        return_value=json.dumps(RESOURCE_JSON)
+    mock_session.post.return_value.__aenter__.return_value.json.return_value = (
+        json.dumps(RESOURCE_JSON)
     )
     mock_session.post.return_value.__aenter__.return_value.status = 202
 
     crp = CreateResourceParams(
         name="resource_name",
-        type=ResourceType.POSTGRES.value,
+        type="postgres",
```
Comment (janelletavares) [inline tests/test_resources.py:83]: what was the issue with the constant?

**Review by janelletavares — APPROVED** [review]
looks reasonable to me

**Comment by ericcheatham** [comment]
Closing this branch as the decision was made to not move forward with this work. IR/unified backend work will be prioritized in accordance

**Pull request closed without merging**